include/exadg/postprocessor/time_control.cpp

```diff
@@ -0,0 +1,173 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+// C/C++
+#include <limits>
+
+// deal.II
+#include <deal.II/base/exceptions.h>
+
+// ExaDG
+#include <exadg/postprocessor/time_control.h>
+#include <exadg/utilities/print_functions.h>
+
+namespace ExaDG
+{
+TimeControlData::EvalType
+get_unsteady_evaluation_type(TimeControlData const & data)
+{
+  bool const interval = (data.trigger_interval > 0.0);
+  bool const timestep = Utilities::is_valid_timestep(data.trigger_every_time_steps);
+
+  // only one of both can be true
+  if(interval && timestep)
+    return TimeControlData::EvalType::Invalid;
+
+  if(interval)
+    return TimeControlData::EvalType::Interval;
+
+  if(timestep)
+    return TimeControlData::EvalType::Timestep;
+
+  return TimeControlData::EvalType::Invalid;
+}
+
+TimeControlData::TimeControlData()
+  : is_active(false),
+    start_time(std::numeric_limits<double>::max()),
+    end_time(std::numeric_limits<double>::max()),
+    trigger_interval(-1.0),
+    trigger_every_time_steps(numbers::invalid_timestep)
+{
+}
+
+void
+TimeControlData::print(dealii::ConditionalOStream & pcout, bool const unsteady) const
+{
+  if(unsteady || is_active)
+  {
+    print_parameter(pcout, "TimeControl start time", start_time);
+    print_parameter(pcout, "TimeControl end_time", end_time);
+    if(get_unsteady_evaluation_type(*this) == TimeControlData::EvalType::Interval)
+      print_parameter(pcout, "TimeControl triggers every interval", trigger_interval);
+    if(get_unsteady_evaluation_type(*this) == TimeControlData::EvalType::Timestep)
+      print_parameter(pcout, "TimeControl triggers every timestep", trigger_every_time_steps);
+  }
+}
+
+TimeControl::TimeControl()
+  : EPSILON(1.0e-10), reset_counter(true), counter(0), end_time_reached(false)
+{
+}
+
+void
+TimeControl::setup(TimeControlData const & time_control_data_in)
+{
+  time_control_data = time_control_data_in;
+}
+
+// If setup() is not called needs_evaluation() will always evaluate to false
```

Comment:
Don't know if we need this comment here. I think it should actually be quite clear.